spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -54,19 +54,25 @@ public ProducerDestination provisionProducerDestination(String name,
 	public ConsumerDestination provisionConsumerDestination(String name, String group,
 			ExtendedConsumerProperties<PubSubConsumerProperties> properties)
 			throws ProvisioningException {
-		if (this.pubSubAdmin.getSubscription(name) == null) {
+		return new PubSubConsumerDestination(name);
+	}
+
+	public String createSubscription(String name, String group,
```

Comment:
Why the extra method?
Shouldn't this work be done in `provisionProducerDestination` like before, but considering the group name instead of just `name`?

Reply:
before this change `provisionConsumerDestination` was called 2 times:

1. `org.springframework.cloud.stream.binder.AbstractMessageChannelBinder#doBindConsumer` 
2. `org.springframework.cloud.gcp.stream.binder.pubsub.PubSubMessageChannelBinder#createConsumerEndpoint`

This resulted in wrong destination names.

Reply:
I Removed the second invocation so that the PubSubConsumerDestination now directly identifies the subscription to be used

Reply:
Not sure if I understand?
We were calling `provisionProducerDestination` once, to automatically create the topic, and `provisionConsumerDestination` once, to create the subscription.

It sounds to me we still need both of those calls.

Reply:
I realised that `provisionProducerDestination` is also called by `AbstractMessageChannelBinder.doBindProducer()`.

Should we also remove that call?